src/components/widgets-settings/dates-picker/dates-picker.directive.coffee

```diff
@@ -109,6 +110,17 @@ module.directive('settingDatesPicker', ($templateCache, $filter, ImpacWidgetsSvc
             keep_today: isToToday()
         }
 
+      scope.onChange = ->
```

Comment:
I don't think this function is needed: you can simply update the standard behaviour, with something like:
```
      scope.applyChanges = ->
        ImpacWidgetsSvc.updateWidgetSettings(w, true) if w
        scope.onChange() if angular.isDefined(scope.onChange)
        scope.changed = false
```